array-filter/filter.js

```diff
@@ -0,0 +1,9 @@
+function filter(array, predicate) {
+  const newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    if (predicate(array[i]) === true) {
```

Comment:
To make this more like the built-in filter use the return values truthiness instead of strict equality to `true`.